backend/coins/eth/blockbook/blockbook.go

```diff
@@ -0,0 +1,264 @@
+func NewBlockbook(chainId string, httpClient *http.Client) *Blockbook {
+	// TODO chainID is not used here, do we have blockbook running for SEPETH as well?
+	return &Blockbook{
+		url:        "https://bb1.shiftcrypto.io/api/",
+		httpClient: httpClient,
+		limiter:    rate.NewLimiter(rate.Limit(callsPerSec), 1),
+		log:        logging.Get().WithField("ETH Client", "Blockbook"),
+	}
+}
+
+func (blockbook *Blockbook) call(ctx context.Context, handler string, params url.Values, result interface{}) error {
+	if err := blockbook.limiter.Wait(ctx); err != nil {
+		return errp.WithStack(err)
+	}
+
+	reqUrl := blockbook.url + handler + "?" + params.Encode()
+
+	response, err := blockbook.httpClient.Get(reqUrl)
+	if err != nil {
+		return errp.WithStack(err)
+	}
+	defer func() { _ = response.Body.Close() }()
+	if response.StatusCode != http.StatusOK {
+		return errp.Newf("expected 200 OK, got %d", response.StatusCode)
+	}
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return errp.WithStack(err)
+	}
+
+	if err := json.Unmarshal(body, result); err != nil {
+		return errp.Newf("unexpected response from blockbook: %s", string(body))
+	}
+
+	return nil
+}
+
+func (blockbook *Blockbook) address(ctx context.Context, account common.Address, params url.Values, result interface{}) error {
+	address := account.Hex()
+
+	if err := blockbook.call(ctx, path.Join("address", address), params, result); err != nil {
+		return errp.WithStack(err)
+	}
+
+	return nil
+}
+
+// Balance implements rpc.Interface.
+func (blockbook *Blockbook) Balance(ctx context.Context, account common.Address) (*big.Int, error) {
+	result := struct {
+		Balance string `json:"balance"`
+	}{}
+
+	if err := blockbook.address(ctx, account, url.Values{}, &result); err != nil {
+		return nil, errp.WithStack(err)
+	}
+
+	balance, ok := new(big.Int).SetString(result.Balance, 10)
+	if !ok {
+		return nil, errp.Newf("could not parse balance %q", result.Balance)
+	}
+	return balance, nil
+
+}
+
+// BlockNumber implements rpc.Interface.
+func (blockbook *Blockbook) BlockNumber(ctx context.Context) (*big.Int, error) {
+	return nil, fmt.Errorf("Not yet implemented")
+}
+
+// PendingNonceAt implements rpc.Interface.
+func (blockbook *Blockbook) PendingNonceAt(ctx context.Context, account common.Address) (uint64, error) {
+	return 0, fmt.Errorf("Not yet implemented")
+}
+
+// prepareTransactions casts to []accounts.Transactions and removes duplicate entries and sets the
+// transaction type (send, receive, send to self) based on the account address.
+func prepareTransactions(
+	isERC20 bool,
+	blockTipHeight *big.Int,
+	isInternal bool,
+	transactions []*Tx, address common.Address) ([]*accounts.TransactionData, error) {
+
+	// TODO figure out if needed. Etherscan.go uses this to compute the num of confirmations.
+	// But numConfirmations is already returned by the API call.
+	_ = blockTipHeight
+
+	_ = isInternal // TODO figure out how to deal with internal txs.
```

Comment:
Please do, my ERC20 accounts don't show any transactions with this PR :sweat_smile: I should probably review once it works for ETH and ERC20, or do you want to separate it into two PRs?